spec/services/container_dashboard_service_spec.rb

```diff
@@ -0,0 +1,87 @@
+  end
+
+  context "providers" do
+    it "returns all providers correctly" do
+      Tempfile.create(['config', 'yml']) do |f|
+        f.write Psych.dump(['ems-type:openshift',
+                            'ems-type:kubernetes',
+                            'ems-type:atomic',
+                            'ems-type:openshift_enterprise',
+                            'ems-type:atomic_enterprise'])
+        f.close
+
+        Vmdb::PermissionStores.configure do |config|
+          config.backend = 'yaml'
+          config.options[:filename] = f.path
+        end
+
+        Vmdb::PermissionStores.initialize!
+      end
+
+      ManageIQ::Providers::Kubernetes::ContainerManager.create(:hostname => "test1.com",
+                                                               :name     => "test1")
+      ManageIQ::Providers::Openshift::ContainerManager.create(:hostname => "test2.com",
+                                                              :name     => "test2")
+      ManageIQ::Providers::OpenshiftEnterprise::ContainerManager.create(:hostname => "test3.com",
+                                                                        :name     => "test3")
+      ManageIQ::Providers::Atomic::ContainerManager.create(:hostname => "test.com4",
+                                                           :name     => "test4")
+      ManageIQ::Providers::AtomicEnterprise::ContainerManager.create(:hostname => "test5.com",
+                                                                     :name     => "test5")
+
+      providers_data = ContainerDashboardService.new(nil, nil).providers
+      expect(providers_data).to eq([{
+                                      :iconClass    => "pficon pficon-kubernetes",
+                                      :count        => 1,
+                                      :id           => :kubernetes,
+                                      :providerType => :Kubernetes
+                                    },
+                                    {
+                                      :iconClass    => "pficon pficon-openshift",
+                                      :count        => 2,
+                                      :id           => :openshift,
+                                      :providerType => :Openshift
+                                    },
+                                    {
+                                      :iconClass    => "pficon pficon-atomic",
+                                      :count        => 2,
+                                      :id           => :atomic,
+                                      :providerType => :Atomic
+                                    }])
+    end
+
+    it "filters providers by provider count and permissions" do
+      Tempfile.create(['config', 'yml']) do |f|
+        f.write Psych.dump(['ems-type:kubernetes',
+                            'ems-type:openshift',
+                            'ems-type:openshift_enterprise'])
+        f.close
```

Comment:
@zeari for the record this is not optimal. Creating temp files should be the last resort. Vmdb::PermissionStore should allow you to load yaml from a string or a file object (not a file path).

Feel free to fix that if you have time ("not in this patch" ™)